smallrye-reactive-messaging-aws-sns/src/main/java/io/smallrye/reactive/messaging/aws/sns/SnsConnector.java

```diff
@@ -0,0 +1,199 @@
+                    .builder()
+                    .topicArn(topicArn)
+                    .message((String) msg.getPayload())
+                    .build();
+            CompletableFuture<PublishResponse> response = snsClient.publish(pr);
+            LOGGER.debug("Message ID {}", response.get().messageId());
+        } catch (InterruptedException | ExecutionException e) {
+            LOGGER.error("Async call interruption happened !!!", e);
+        }
+    }
+
+    @Override
+    public PublisherBuilder<? extends Message<?>> getPublisherBuilder(Config config) {
+
+        String topicName = getTopicName(config);
+        Integer port = config.getOptionalValue("port", Integer.class).orElse(8080);
+        boolean broadcast = config.getOptionalValue("broadcast", Boolean.class).orElse(false);
+        int initialDelay = config.getOptionalValue("initDelay", Integer.class).orElse(2000);
+        SnsVerticle snsVerticle = new SnsVerticle(getAppURL(), topicName, port);
+        vertx.deployVerticle(snsVerticle);
+
+        Flowable<SnsMessage<String>> flowable = Flowable.fromCallable(() -> {
+
+            SnsMessage<String> snsMsg = null;
+            try {
+                LOGGER.trace("Polling message");
+                //It uses blockingQ.take() method so it will block until there is 
+                //an item available. Thats why in repeatWhen it is only 1 millisecond.
+                snsMsg = snsVerticle.pollMsg();
+                Objects.requireNonNull(snsMsg, "Null message has been consumed from Q");
+            } catch (InterruptedException e) {
+                LOGGER.error("Polling interrupted", e);
+                throw e;
+            }
+            return snsMsg;
+        }).repeatWhen(o -> o.concatMap(v -> Flowable.timer(1, TimeUnit.MILLISECONDS)))
+                .delaySubscription(initialDelay, TimeUnit.MILLISECONDS)
+                .subscribeOn(scheduler);
+
+        PublisherBuilder<? extends Message<?>> builder = ReactiveStreams.fromPublisher(flowable);
+
+        return broadcast ? builder.via(MulticastProcessor.create()) : builder;
+    }
+
+    /**
+     * Retrieve topic name and throws exception if it is not configured.
+     * 
+     * @param config microprofile config instance.
+     * @return topic name.
+     */
+    private String getTopicName(Config config) {
+        return config.getOptionalValue("address", String.class)
+                .orElseGet(
+                        () -> config.getOptionalValue("topic-name", String.class)
+                                .orElseThrow(() -> new IllegalArgumentException("Address/topic-name must be set")));
+    }
+
+    /**
+     * Retrieve App URL and throws exception if it is not configured.
+     * 
+     * @param config microprofile config instance.
+     * @return application URL accessible by AWS SNS.
+     */
+    private String getAppURL() {
+        return appURL.orElseThrow(() -> new IllegalArgumentException("App URL must be set"));
```

Comment:
Should the AppURL also configurable per channel? Imagine an application connected to 2 SNS endpoints.

Reply:
The App URL is your application URL which contains SNS Verticle. Usually you will have one SNS endpoint per account.

Reply:
What do you think ? SNS endpoint will be one for sure because it is related to your AWS account. shall I make it per channel or as general ? from my point of view, I think it should be global config.

Reply:
Yes, It should be global.